src/screens/Settings/LabSettings/FeatureGateDialog.tsx

```diff
@@ -0,0 +1,215 @@
+import {Button, ButtonIcon, ButtonText} from '#/components/Button'
+import * as Dialog from '#/components/Dialog'
+import {Divider} from '#/components/Divider'
+import * as Toggle from '#/components/forms/Toggle'
+import {
+  EmojiSad_Stroke2_Corner0_Rounded as EmojiSad,
+  EmojiSmile_Stroke2_Corner0_Rounded as EmojiSmile,
+} from '#/components/icons/Emoji'
+import {Loader} from '#/components/Loader'
+import {P, Text} from '#/components/Typography'
+
+enum FeedbackToken {
+  LoveIt = 'love-it',
+  HateIt = 'hate-it',
+  Useful = 'useful',
+  MakesAppLooksBetter = 'makes-app-look-better',
+  MakesThingsEasier = 'makes-things-easier',
+  PromisingButNeedsWork = 'promising-but-needs-work',
+  RemovesImportantFeatures = 'removes-important-features',
+  ConfusedMe = 'confused-me',
+  BrokeMyWillToLive = 'broke-my-will-to-live',
+}
+
+export function FeatureGateDialog({
+  control,
+  gate,
+}: {
+  control: Dialog.DialogOuterProps['control']
+  gate: Gate
+}) {
+  const {_} = useLingui()
+  const t = useTheme()
+  const descriptions = useGateDescriptions()
+  const gateApi = useGate()
+  const setGateApi = useSetLocalGateOverride()
+
+  const [originalEnabled] = useState(() => gateApi(gate))
+  const [enabled, setEnabled] = useState(originalEnabled)
+  const [isRestarting, setIsRestarting] = useState(false)
+  const desc = descriptions[gate]
+
+  const onToggleGate = (v: boolean) => {
+    logger.metric('featureGate:override', {gate, enabled: v})
+    setGateApi(gate, v)
+    setEnabled(v)
+  }
+
+  if (!desc) {
+    return null
+  }
+
+  return (
+    <Dialog.Outer control={control}>
+      <Dialog.Handle />
+      <Dialog.ScrollableInner
+        accessibilityDescribedBy="dialog-description"
+        accessibilityLabelledBy="dialog-title">
+        <View style={[a.relative, a.gap_md, a.w_full]}>
+          <Text
+            nativeID="dialog-title"
+            style={[a.text_2xl, a.font_bold, t.atoms.text]}>
+            {desc.title}
+          </Text>
+
+          <P nativeID="dialog-description">{desc.description}</P>
+
+          <View
+            style={[a.rounded_sm, t.atoms.bg_contrast_25, a.px_md, a.py_sm]}>
+            <Toggle.Item
+              name="quoteposts"
+              type="checkbox"
+              label={_(msg`Tap to toggle this experiment.`)}
+              value={enabled}
+              onChange={onToggleGate}
+              style={[a.justify_between]}>
+              <Text style={[t.atoms.text_contrast_high]}>
+                <Trans>Enabled on this device</Trans>
+              </Text>
+              <Toggle.Switch />
+            </Toggle.Item>
+          </View>
+
+          {enabled !== originalEnabled ? (
+            <Button
+              variant="solid"
+              color="primary"
+              size="large"
+              onPress={() => {
+                setIsRestarting(true)
+                if (isWeb) {
+                  /* @ts-ignore web only */
+                  navigation.reload()
```

Comment:
`window.navigation` is not available on Firefox/Safari. better to use the old fashioned `location.reload()`

```suggestion
                  location.reload()
```